brain/circuit.h

```diff
@@ -120,15 +120,55 @@ class Circuit : public boost::noncopyable
 
     /** @return The local to world transformations of the given cells. */
     BRAIN_API Matrix4fs getTransforms( const GIDSet& gids ) const;
+
     /** @return The local to world rotation of the given cells. */
     BRAIN_API Quaternionfs getRotations( const GIDSet& gids ) const;
+
     /** @return The number of neurons in the circuit. */
     BRAIN_API size_t getNumNeurons() const;
 
+    /**
+     * Access all afferent synapses of the given GIDs.
+     *
+     * @param gids the gids to load afferent synapses for
+     * @param prefetch which synapse data to load on SynapsesStream.read()
+     * @return object to stream synapses data
+     */
+    BRAIN_API SynapsesStream getAfferentSynapses( const GIDSet& gids,
+                                      SynapsePrefetch prefetch = SP_ALL ) const;
+
+    /**
+     * Access all efferent synapses of the given GIDs.
+     *
+     * @param gids the gids to load efferent synapses for
+     * @param prefetch which synapse data to load on SynapsesStream.read()
+     * @return object to stream synapses data
+     */
+    BRAIN_API SynapsesStream getEfferentSynapses( const GIDSet& gids,
+                                      SynapsePrefetch prefetch = SP_ALL ) const;
+
+    /**
+     * Access all synapses along the projection from the pre- to the postGIDs.
+     *
+     * @param preGIDs the gids to load the efferent synapses for
+     * @param postGIDs the gids to load the afferent synapses for
+     * @param prefetch which synapse data to load on SynapsesStream.read()
+     * @return object to stream synapses data
+     */
+    BRAIN_API SynapsesStream getProjectedSynapses( const GIDSet& preGIDs,
+                                                   const GIDSet& postGIDs,
+                                      SynapsePrefetch prefetch = SP_ALL ) const;
+
     class Impl; //!< @internal, public for inheritance MVD2/3 impls
 
 private:
-    Impl* _impl;
+    friend class Synapses;
+
+    const brion::SynapseSummary& getSynapseSummary() const;
+    const brion::Synapse& getSynapseAttributes( bool afferent ) const;
+    const brion::Synapse* getSynapseExtra() const;
+    const brion::Synapse& getSynapsePositions( bool afferent ) const;
```

Comment:
Move these functions to the Impl object, that will save a lot of _impl-> text in the implementation and you can change the implementation later without touching this file.